packages/mui-codemod/src/deprecations/divider-props/divider-props.js

```diff
@@ -12,43 +12,52 @@ export default function transformer(file, api, options) {
   const printOptions = options.printOptions;
 
   findComponentJSX(j, { root, componentName: 'Divider' }, (elementPath) => {
-    const hasLightProp =
-      elementPath.node.openingElement.attributes.findIndex(
-        (attr) => attr.type === 'JSXAttribute' && attr.name.name === 'light',
-      ) !== -1;
-
-    if (hasLightProp) {
-      elementPath.node.openingElement.attributes =
-        elementPath.node.openingElement.attributes.filter((attr) => {
-          if (attr.type === 'JSXAttribute' && attr.name.name === 'light') {
-            return false;
-          }
-          return true;
-        });
+    const lightProp = elementPath.node.openingElement.attributes.find(
+      (attr) => attr.type === 'JSXAttribute' && attr.name.name === 'light',
+    );
 
-      const sxIndex = elementPath.node.openingElement.attributes.findIndex(
-        (attr) => attr.type === 'JSXAttribute' && attr.name.name === 'sx',
-      );
-      if (sxIndex === -1) {
-        appendAttribute(j, {
-          target: elementPath.node,
-          attributeName: 'sx',
-          expression: j.objectExpression([
-            j.objectProperty(j.identifier('opacity'), j.literal('0.6')),
-          ]),
-        });
-      } else {
-        const opacityIndex = elementPath.node.openingElement.attributes[
-          sxIndex
-        ].value.expression.properties.findIndex((key) => key.key.name === 'opacity');
-
-        if (opacityIndex === -1) {
-          assignObject(j, {
-            target: elementPath.node.openingElement.attributes[sxIndex],
-            key: 'opacity',
-            expression: j.literal('0.6'),
-          });
+    if (!lightProp) {
+      return;
+    }
+
+    elementPath.node.openingElement.attributes = elementPath.node.openingElement.attributes.filter(
+      (attr) => {
+        if (attr.type === 'JSXAttribute' && attr.name.name === 'light') {
+          return false;
         }
+        return true;
+      },
+    );
+
+    const isLightPropTruthy =
+      !!lightProp.value?.expression.value || lightProp.value?.expression.value === undefined;
+
+    if (!isLightPropTruthy) {
```

Comment:
If `light={false}` we already remove the prop so we don't continue with adding anything to the `sx` prop.